### DISCUSSION TITLE
[ 搜索结果重复 ] 同一个句子在搜索结果中多次出现

### DISCUSSION THREAD

**Discussion by luciffer118** [vuepress-theme-hope/vuepress-theme-hope#3393] [Q&A] [answered]
想请教一下怎样配置才能避免结果只出现一次？
如图，同一个句子中，有多个分词结果被匹配。然后就都显示在界面上了，很混乱。
![image](https://github.com/vuepress-theme-hope/vuepress-theme-hope/assets/142007421/88d962bf-5365-45a7-8331-f9a32c6f0541)
![image](https://github.com/vuepress-theme-hope/vuepress-theme-hope/assets/142007421/9677ea2a-bea2-4029-ac3d-aa8a71bdca6a)

**Comment by Mister-Hope** [comment, marked as answer]
客户端搜索本来能力就有限。

我们以一个著名的 “南京市长江大桥” 来举例子，客户端搜索不像搜索引擎有语义识别能力，它不知道当搜索结果不包含南京市长江大桥时，核心词是 “南京市 长江大桥”还是“南京市长 江大桥”，因此在构建期间，所有你可能用到的分词组合，都会被录入。

这包含第二个例子售电站，分词软件认为它包含售电和电站两个词组，因此 `售电` `站` `站点`(售电站点的分词结果) `售` `电站` 等词都是有效的。

如果你想进一步优化这些搜索结果，你可以为 nodejs-jieba 额外添加“售电站”词汇加入字典，也可以调节 BM25 参数，对搜索词的权重和匹配进行优化。https://mister-hope.github.io/slimsearch/interfaces/BM25Params.html

**Reply by Mister-Hope** [reply]
为了搜索结果的精确，你应该自行提供专业用语或文中非常见词的词典，例如：https://github.com/inNENU/resource/blob/main/site/.vuepress/words

**Reply by Mister-Hope** [reply]
https://github.com/vuepress-theme-hope/vuepress-theme-hope/commit/186e84eb8a1c582bd42388cc65deabf27a1fb407
此 commit 改进了显示的问题，但它并不能改进排序之类的实际问题
